test/blackbox-tests/test-cases/pkg/opam-var/opam-var-global.t

```diff
@@ -0,0 +1,102 @@
+  $ . ./opam-var-helpers.sh
+
+Here we test the translation and implementation of global opam variables.
+This also includes switch specific variables since these are all trivial for Dune.
+opam-version
+  $ test opam-version
+  (build
+   (run echo %{pkg-self:opam-version}))
+  File "dune.lock/testpkg.pkg", line 4, characters 11-35:
+  4 |  (run echo %{pkg-self:opam-version}))
+                 ^^^^^^^^^^^^^^^^^^^^^^^^
+  Error: Unknown macro %{pkg-self:..}
+  [1]
+jobs
+  $ test_lockfile jobs 
+  (build
+   (run echo %{jobs}))
+make
+  $ where_is_make="$(dirname $(which make))" 
+  > BUILD_PATH_PREFIX_MAP="MAKE_DIR=$where_is_make:$BUILD_PATH_PREFIX_MAP"
+  > test make 
+  (build
+   (run echo %{make}))
+  MAKE_DIR/make
+arch
+  $ ARCH=$(uname -m)
+  > test arch 2>&1 | sed "s/^$ARCH/ARCH/g"  
+  (build
+   (run echo %{arch}))
+  ARCH
+os
+  $ OS=$(uname -s | awk '{print tolower($0)}')
+  > test os 2>&1 | sed "s/^$OS/OS/g"
+  (build
+   (run echo %{os}))
+  OS
+os-distribution
+TODO: seems to be broken (on NixOS)
```

Comment:
This is broken.

Reply:
You should describe the broken behavior on NixOS inside the test.